### DISCUSSION TITLE
Live Word Count?

### DISCUSSION THREAD

**Discussion by alandove** [geany/geany#3340] [Q&A]
Hey, folks:

Geany is so close to being my perfect text editor. There's one thing I can't seem to find, though: some way to get a live updating word count. I can use the Word Count tool of course, but that has to be clicked each time I want to check the count. What I want is some way to show a continuously updated count, ideally in the status bar at the bottom where the other stats are, but I'd settle for having it update in the Word Count box once it's open. 

Is there any way to get this to work, or is it a missing feature?

**Comment by elextr** [comment]
The character (actually code point I think)/word/line counts are only calculated when the menu item is selected.  Not sure why it doesn't have a keybinding (that I could find).

The dialog is created each time the menu item is activated, it is not updated.

Line count is available on the status bar because it is available from Scintilla.

**Reply by alandove** [reply]
A keybinding - or at least the option to create one - for the word count would definitely be a step in the right direction. 

I gather Scintilla has a concept of words, but I guess someone would still have to write a word count algorithm in Geany to get that information into the status bar.
